Amazon.QLDB.Driver.Tests/AsyncQldbDriverTests.cs

```diff
@@ -0,0 +1,361 @@
+            {
+                ExecuteStatement = new ExecuteStatementResult
+                {
+                    FirstPage = new Page
+                    {
+                        NextPageToken = null
+                    }
+                },
+                ResponseMetadata = new ResponseMetadata
+                {
+                    RequestId = "testId"
+                }
+            };
+        
+            var sendCommandResponseCommit = new SendCommandResponse
+            {
+                CommitTransaction = new CommitTransactionResult
+                {
+                    TransactionId = TestTransactionId,
+                    CommitDigest = new MemoryStream(h1.Hash),
+                },
+                ResponseMetadata = new ResponseMetadata
+                {
+                    RequestId = "testId"
+                }
+            };
+            var mockCreator = new Mock<Func<CancellationToken, Task<Session>>>();
+            var mockSession = new Mock<Session>(null, null, null, null, null);
+        
+            mockSession.Setup(x => x.StartTransactionAsync(default)).ReturnsAsync(sendCommandResponseWithStartSession.StartTransaction);
+            mockSession.SetupSequence(x => x.ExecuteStatementAsync(It.IsAny<String>(), It.IsAny<String>(), It.IsAny<List<IIonValue>>(), default))
+                .Throws(exception)
+                .Throws(exception)
+                .ReturnsAsync(sendCommandResponseExecute.ExecuteStatement);
+            mockSession.Setup(x => x.CommitTransactionAsync(It.IsAny<String>(), It.IsAny<MemoryStream>(), default))
+                .ReturnsAsync(sendCommandResponseCommit.CommitTransaction);
+        
+            mockCreator.Setup((x) => x(default)).ReturnsAsync(mockSession.Object);
+        
+            var driver = new AsyncQldbDriver(
+                new AsyncSessionPool(mockCreator.Object, AsyncQldbDriverBuilder.CreateDefaultRetryHandler(NullLogger.Instance), 4, NullLogger.Instance));
+        
+            try
+            {
+                await driver.Execute(txn => txn.Execute(statement));
+            }
+            catch (Exception e)
+            {
+                Assert.IsTrue(expectThrow);
+                Assert.AreEqual(exception, e);
+                return;
+            }
+        
+            Assert.IsFalse(expectThrow);
+        }
+        
+        internal static ValueHolder CreateValueHolder(IIonValue ionValue)
+        {
+            MemoryStream stream = new MemoryStream();
+            using (var writer = IonBinaryWriterBuilder.Build(stream))
+            {
+                ionValue.WriteTo(writer);
+                writer.Finish();
+            }
+        
+            var valueHolder = new ValueHolder
+            {
+                IonBinary = new MemoryStream(stream.GetWrittenBuffer())
+            };
+        
+            return valueHolder;
+        }
+        
+        internal static IEnumerable<object[]> CreateDriverExceptions()
```

Comment:
Ditto

Reply:
The dynamic data test method seems to require it be defined in the same file rather than a utility file.